File: 2024_01/TP_1/palindromo.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int verificar_palindromo(char *str){
    int i = 0;
    int j = strlen(str) - 1;

    while(i < j){
        if(str[i] != str[j]){
            return 0;
        }

        i++;
        j--;
    }

    return 1;
}

int main(){
    char line[400];
    int continua = 1;

    while(continua){
        fgets(line, sizeof(line), stdin);
        line[strcspn(line, "\n")] = '\0';

        if(strcmp(line, "FIM") == 0){
            continua = 0;
        } else{
            if(verificar_palindromo(line) == 1){
                printf("SIM\n");
            } else{
                printf("NAO\n");
            }
        }
    }

 return 0;
}

File: 2024_01/TP_1/palindromo.java
import java.util.Scanner;

public class palindromo{
    private static boolean recursive(String str, int I, int F){
        if(I >= F){
            return true;
        }

        if(str.charAt(I) != str.charAt(F)){
            return false;
        }

      return recursive(str, I + 1, F);
    }

    public static boolean Palindromo(String str){
        return recursive(str, 0, str.length());
    }

    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        String line;

        while (!(line = sc.nextLine()).equals("FIM")) {
            if (Palindromo(line)) {
                System.out.println("SIM");
            } else {
                System.out.println("NAO");
            }
        }

        sc.close();
    }
}
